### ISSUE TITLE
Add a way to record features extracted during optimisations

### ISSUE THREAD

**Issue by wvangeit** [BlueBrain/BluePyOpt#215]
At the moment the scores of the parameters are returned to the master process. We should also have a way to e.g. get the raw efeature values out. This would be useful for sensitivity analysis e.g.

**Comment by elisabettai** [comment]
It would be very nice to have this enhancement. For example to analyse how features extracted from good models compare to the experimental ones.

**Comment by wvangeit** [comment]
I will see when I have time to implement this. It won't be trivial, because it involves some changes to the current user-facing API. I'll try to implement in a way so that it doesn't affect existing scripts.

**Comment by elisabettai** [comment]
Ok, I see. Another solution would be to re-run models (>10, potentially hundreds parameter combinations). Maybe this can be efficiently done in parallel (using multiprocessing or ipyparallel), by re-using existing code.

**Comment by wvangeit** [comment]
Yes, for now that might be the best solution. You could use the ipyparallel map function for that. You would only have to pass a function that returns the feature values instead of the scores.
Another option is to subclass response or score calculator classes and let them write out the feature values they see to files.

**Comment by wvangeit** [comment]
@DrTaDa, could you have a look at this. It's an often requested feature. Not urgent though.

**Comment by DrTaDa** [comment]
Does #350 answers entirely to what was requested here ?
